packages/features/config-context.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+
+import { ChildrenProps, ReactEvent } from '@packages/utils/react';
+import createCtx from '@packages/utils/createCtx';
+import { FormState } from '@packages/entities/config-modal';
+
+interface ConfigActions {
+  readonly onConfigChange: (event: ReactEvent) => void;
+}
+
+interface ConfigStates {
+  readonly notionApiKey: string;
+  readonly electionDatabaseId: string;
+  readonly resultsDatabaseId: string;
+}
+
+const [useConfigActions, ConfigActionsProvider] =
+  createCtx<ConfigActions>('ConfigActionsCtx');
+const [useConfigStates, ConfigStatesProvider] =
+  createCtx<ConfigStates>('ConfigStatesCtx');
+
+function ConfigProvider({ children }: ChildrenProps) {
+  const [formState, setFormState] = useState<FormState>({
+    notionApiKey: '',
+    electionDatabaseId: '',
+    resultsDatabaseId: '',
+  });
+
+  const onConfigChange = (event: ReactEvent) => {
+    setFormState({
+      ...formState,
+      [event.target.name]: event.target.value,
+    });
+  };
```

Comment:
As we have states and actions in the same context, it'd be better to use a `useCallback` on the actions
```suggestion
  const onConfigChange = useCallback((event: ReactEvent) => {
    setFormState(formState => ({
      ...formState,
      [event.target.name]: event.target.value,
    }));
  }, [setFormState]);
```

this will avoid recreating the actions whenever a state change happens

Reply:
You mean that, when there is a state change the code will re-instantiate the exported actions ? Something like that?

Updating it shortly

Reply:
Done too, thx for the review